src/core/patterns.js

```diff
@@ -59,3 +59,61 @@ foam.CLASS({
     }
   ]
 });
+
+/** Causes an class to pool its instances. create() will pull from the pool,
+ and destroy() will return instances to the pool. Object pools can be found
+ in <code>foam.__objectPools__</code>. */
+foam.CLASS({
+  package: 'foam.pattern',
+  name: 'Pooled',
+  axioms: [ foam.pattern.Singleton.create() ],
+
+  methods: [
+    function installInClass(cls) {
+      // Keeping the object pools in an accessible location allows them
+      // to be cleared out.
+      if ( ! foam.__objectPools__ ) { foam.__objectPools__ = {}; }
```

Comment:
Ah right, of course, forgot they only contain destroyed objects.

I see what you're saying, and I'm not sure which would be better in the long run.  I still think it's a more useful property to be able to garbage collect a class and its associated pool, it just feels better componentized than creating a special global value.

Actually in regards to the global, what of instead of putting __objectPools__ on "foam" add it as a property to foam.pattern.Pooled.  Pooled is afterall a singleton, so seems a sensible place to store "global" values only used by this pattern without mucking with the global foam object.